### PR TITLE
Prefer $XDG_DATA_HOME for datastore

### LINKED ISSUE #859: hashira does not respect `XDG Base Directory`
README said https://github.com/pankona/hashira/blob/3d948785821094eff42385084300cf5064a26cef/README.md#L73-L74

After executing hashira-cui, it generates `db` into the path.
As I understand it, `$HOME/.config` is the default of `$XDG_CONFIG_HOME`.
The db is not a config, it looks a data, data should be put one of the `$XDG_DATA_*`, `$XDG_STATE_*`, `$XDG_CACHE_*` 🤔 

ref: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

### PR DESCRIPTION
**Pull request by kachick**
Fixes #859 

As I understand https://pankona.github.io/hashira/tables.html, `hashira/db` stores data, it is not the config. Is this correct?
This PR maintains the current behavior if the old directory exists.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module github.com/pankona/hashira
 
 require (
+	github.com/adrg/xdg v0.4.0
```
Comment (pankona) [inline go.mod:4]: It seems this library is not maintained actively. Can you find another package or is it better to implement this by ourself?

Comment (kachick) [reply go.mod:4]: This library looks used in these few years. And I can understand it is not nessesery to change from the purpose 🤔.

File: xdg/xdg_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			"XDG_DATA_HOME":   "",
+			"XDG_CONFIG_HOME": "/home/hashirafan/.config",
+			"XDG_STATE_HOME":  "/home/hashirafan/.local/state",
+			"XDG_CACHE_HOME":  "/home/hashirafan/.cache",
+			"XDG_DATA_DIRS":   "/usr/local/share:/usr/share",
+		}},
+	}
+
+	for _, tc := range testCases {
+		setup(t, tc.env)
```
Comment (kachick) [inline xdg/xdg_test.go:43]: Is this not unsetting the defined env after a test case? 🤔

**Review by pankona — APPROVED** [review]
LGTM! Thank you!

**Comment by kachick** [comment]
Thank you! ☸️

**Pull request merged**